service/src/main/java/org/jboss/sbomer/service/feature/sbom/errata/event/release/ReleaseAdvisoryEventsListener.java

```diff
@@ -0,0 +1,471 @@
+import org.jboss.sbomer.core.dto.v1beta1.V1Beta1RequestRecord;
+import org.jboss.sbomer.core.errors.ApplicationException;
+import org.jboss.sbomer.core.features.sbom.enums.GenerationRequestType;
+import org.jboss.sbomer.core.features.sbom.utils.SbomUtils;
+import org.jboss.sbomer.service.feature.sbom.errata.ErrataClient;
+import org.jboss.sbomer.service.feature.sbom.errata.dto.Errata.Details;
+import org.jboss.sbomer.service.feature.sbom.errata.dto.ErrataBuildList.BuildItem;
+import org.jboss.sbomer.service.feature.sbom.errata.dto.ErrataBuildList.ProductVersionEntry;
+import org.jboss.sbomer.service.feature.sbom.errata.event.AdvisoryEventUtils;
+import org.jboss.sbomer.service.feature.sbom.k8s.model.SbomGenerationStatus;
+import org.jboss.sbomer.service.feature.sbom.model.Sbom;
+import org.jboss.sbomer.service.feature.sbom.model.SbomGenerationRequest;
+import org.jboss.sbomer.service.feature.sbom.pyxis.PyxisClient;
+import org.jboss.sbomer.service.feature.sbom.pyxis.dto.PyxisRepository;
+import org.jboss.sbomer.service.feature.sbom.pyxis.dto.PyxisRepositoryDetails;
+import org.jboss.sbomer.service.feature.sbom.pyxis.dto.RepositoryCoordinates;
+import org.jboss.sbomer.service.feature.sbom.service.SbomGenerationRequestRepository;
+import org.jboss.sbomer.service.feature.sbom.service.SbomService;
+import org.jboss.sbomer.service.stats.StatsService;
+
+import io.quarkus.narayana.jta.QuarkusTransaction;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.ObservesAsync;
+import jakarta.inject.Inject;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@ApplicationScoped
+@Slf4j
+public class ReleaseAdvisoryEventsListener {
+
+    // Set the long transaction imeout to 10 mins
+    private static final int INCREASED_TIMEOUT_SEC = 600;
+
+    @Inject
+    @RestClient
+    @Setter
+    ErrataClient errataClient;
+
+    @Inject
+    @RestClient
+    @Setter
+    PyxisClient pyxisClient;
+
+    @Inject
+    @Setter
+    SbomService sbomService;
+
+    @Inject
+    @Setter
+    StatsService statsService;
+
+    @Inject
+    @Setter
+    SbomGenerationRequestRepository generationRequestRepository;
+
+    private static final String NVR_STANDARD_SEPARATOR = "-";
+
+    public void onReleaseAdvisoryEvent(@ObservesAsync AdvisoryReleaseEvent event) {
+        log.debug("Event received for advisory release ...");
+
+        if (event.getErratum().getDetails().get().getContentTypes().contains("docker")) {
+            releaseManifestsForDockerBuilds(event);
+        } else {
+            releaseManifestsForRPMBuilds();
+        }
+    }
+
+    protected void releaseManifestsForRPMBuilds() {
+        throw new ApplicationException("**** NOT IMPLEMENTED ****");
+    }
+
+    protected void releaseManifestsForDockerBuilds(AdvisoryReleaseEvent event) {
+
+        Details details = event.getErratum().getDetails().get();
+        V1Beta1RequestRecord advisoryManifestsRecord = event.getLatestAdvisoryManifestsRecord();
```

Comment:
Where can I find this method? I think I'm blind :)

Reply:
The `AdvisoryReleaseEvent` contains this value already, and the getters are generated by lombok? https://github.com/project-ncl/sbomer/pull/1045/files/8c8dac2ead6a37adc7e3560771cee00a05c57d94#diff-794872ceb8102f4aaa54c141386f6de66ae011006d26d27771180444653dc1ceR37

Reply:
Thanks. Not sure why I was thinking that this must have been an actual method. And I was stuck at this :)